dask/dataframe/tests/test_groupby.py

```diff
@@ -3555,11 +3555,11 @@ def test_groupby_numeric_only_supported(func, numeric_only):
         with ctx:
             expected = getattr(pdf.groupby("ints"), func)(**kwargs)
         successful_compute = True
-    except TypeError as e:
+    except TypeError:
         # Make sure dask and pandas raise the same error message
         # We raise the error on _meta_nonempty, actual element may differ
         ctx = pytest.raises(
-            TypeError, match=f"{str(e)[:-5]}|does not support reduction"
+            TypeError, match="Cannot convert|could not convert|does not support"
```

Comment:
Hmm it looks like the actual data is included in the error message here

```python
FAILED dask/dataframe/tests/test_groupby.py::test_groupby_numeric_only_supported[disk-None-median] - AssertionError: Regex pattern did not match.
 Regex: "Cannot convert ['q' 'c'] to nu|does not support reduction"
 Input: "Cannot convert ['foo' 'foo'] to numeric"
```

Note that since we're raising on `_meta_nonempty` here `"foo"` ends up in the error message, which isn't the most informative to users since that's due to our internal non-empty meta generation logic. It'd be good to revisit this (e.g. can we raise on `_meta` instead? If so, is the error message better?)

This is outside the scope of this PR -- just wanted to raise this as a follow-up item

Reply:
Yeah I thought about this too, but couldn't find a better solution yet. It will try to compute the result on object or string columns and will succeed if you have integers in your object/string column. This is one of the reasons why it will succeed with an empty object as well, because this can actually be cast to float. I am trying to push towards an error message the excludes the values though, so this would solve the problem